substrate/frame/broker/src/migration.rs

```diff
@@ -156,6 +162,244 @@ mod v3 {
+	impl<T: Config, BlockConversion: BlockToRelayHeightConversion<T>> UncheckedOnRuntimeUpgrade
+		for MigrateToV4Impl<T, BlockConversion>
+	{
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			let (interlude_length, configuration_leadin_length) =
+				if let Some(config_record) = v3::Configuration::<T>::get() {
+					(config_record.interlude_length, config_record.leadin_length)
+				} else {
+					((0 as u32).into(), (0 as u32).into())
+				};
+
+			let updated_interlude_length: RelayBlockNumberOf<T> =
+				BlockConversion::convert_block_length_to_relay_length(interlude_length);
+			let updated_leadin_length: RelayBlockNumberOf<T> =
+				BlockConversion::convert_block_length_to_relay_length(configuration_leadin_length);
+			log::info!(target: LOG_TARGET, "Configuration Pre-Migration: Interlude Length {:?}->{:?} Leadin Length {:?}->{:?}", interlude_length, updated_interlude_length, configuration_leadin_length, updated_leadin_length);
+
+			let (sale_start, sale_info_leadin_length) =
+				if let Some(sale_info_record) = v3::SaleInfo::<T>::get() {
+					(sale_info_record.sale_start, sale_info_record.leadin_length)
+				} else {
+					((0 as u32).into(), (0 as u32).into())
+				};
+
+			let updated_sale_start: RelayBlockNumberOf<T> =
+				BlockConversion::convert_block_number_to_relay_height(sale_start);
+			let updated_sale_info_leadin_length: RelayBlockNumberOf<T> =
+				BlockConversion::convert_block_length_to_relay_length(sale_info_leadin_length);
+			log::info!(target: LOG_TARGET, "SaleInfo Pre-Migration: Sale Start {:?}->{:?} Interlude Length {:?}->{:?}", sale_start, updated_sale_start, sale_info_leadin_length, updated_sale_info_leadin_length);
+
+			Ok((interlude_length, configuration_leadin_length, sale_start, sale_info_leadin_length)
+				.encode())
+		}
+
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			let mut weight = T::DbWeight::get().reads(1);
+
+			if let Some(config_record) = v3::Configuration::<T>::take() {
+				log::info!(target: LOG_TARGET, "migrating Configuration record");
+
+				let updated_interlude_length: RelayBlockNumberOf<T> =
+					BlockConversion::convert_block_length_to_relay_length(
+						config_record.interlude_length,
+					);
+				let updated_leadin_length: RelayBlockNumberOf<T> =
+					BlockConversion::convert_block_length_to_relay_length(
+						config_record.leadin_length,
+					);
+
+				let updated_config_record = ConfigRecord {
+					interlude_length: updated_interlude_length,
+					leadin_length: updated_leadin_length,
+					advance_notice: config_record.advance_notice,
+					region_length: config_record.region_length,
+					ideal_bulk_proportion: config_record.ideal_bulk_proportion,
+					limit_cores_offered: config_record.limit_cores_offered,
+					renewal_bump: config_record.renewal_bump,
+					contribution_timeout: config_record.contribution_timeout,
+				};
```

Comment:
```suggestion
				let updated_config_record = ConfigRecord {
					interlude_length: updated_interlude_length,
					leadin_length: updated_leadin_length,
					..config_record
				};
```

Reply:
It's a compile error, because these are distinct types due to migration, although they have similar fields